pkg/provisioner/ironic/ironic.go

```diff
@@ -1212,6 +1245,21 @@ func (p *ironicProvisioner) Provision(hostConf provisioner.HostConfigData) (resu
 		return transientError(provisioner.NeedsRegistration)
 	}
 
+	if p.host.Spec.AutomatedCleaningMode == "None" {
```

Comment:
What is this "None" ? You should also consider the empty string (or nil pointer if you make it a pointer)

Reply:
Oh sorry, this should be `Disabled`. Forgot to replace it.

Reply:
Fixed it. Thanks.